src/Data/Factory.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Data;
+
+/**
+ * Builds data types.
```

Comment:
If you can think of a simple example and add that to the interface along with some examples, that would help a lot.